### PR TITLE
Add ctf_enum_auto() for autoincrementing enumeration values

### PR DESCRIPTION
**Pull request by eepp**

### CODE REVIEW

File: include/lttng/ust-events.h [hunk 1]
```diff
@@ -110,7 +110,12 @@ struct lttng_enum_value {
 struct lttng_enum_entry {
 	struct lttng_enum_value start, end; /* start and end are inclusive */
 	const char *string;
-	char padding[LTTNG_UST_ENUM_ENTRY_PADDING];
+	union {
+		struct {
+			unsigned int is_auto:1;
```
Comment (compudj) [inline include/lttng/ust-events.h]: I'm tempted to use uint8_t here instead. Bitfields may differ between compiler implementations, and I'm tempted not to use them for an ABI between probe providers and UST.

Comment (eepp) [inline include/lttng/ust-events.h]: I'm using this because it's used in `struct lttng_enum_value` just above.

Comment (compudj) [inline include/lttng/ust-events.h]: I should probably not have used a bitfield there then. Let's not spread this.

Comment (eepp) [inline include/lttng/ust-events.h]: How about an `unsigned int` with option bits? I don't want to waste 7 bits, it's too hard.

Comment (compudj) [inline include/lttng/ust-events.h]: works for me. A "flags" unsigned int. The option bits should be defined as e.g.

enum ...myflags {
   MYFLAG_A = (1U << 0),
}

**Comment by eepp** [comment]
How's 9054ed2?

**Pull request closed without merging**